.github/workflows/pipeline.yml

```diff
@@ -0,0 +1,65 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  run_unit_tests:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '8.x'
+
+    - name: Install dependencies
+      run: dotnet restore
+
+    - name: Build solution
+      run: dotnet build
+
+    - name: Run tests
+      run: |
+        dotnet test Chefster.Tests/Chefster.Tests.csproj
+
+  deploy_to_ec2:
+    needs: run_unit_tests
+    if: github.ref == 'refs/heads/master' && github.event_name == 'push'
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '8.x'
+
+    - name: Build project
+      run: dotnet publish -c Release -o app
+
+    - name: Deploy to EC2
+      env:
+        PRIVATE_KEY: ${{ secrets.EC2_PRIVATE_KEY }}
+        HOST_NAME: ${{ secrets.EC2_HOST_NAME }}
+        USER_NAME: ${{ secrets.EC2_USER_NAME }}
+      run: |
+        echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+        ssh -i private_key -o StrictHostKeyChecking=no ${USER_NAME}@${HOST_NAME} `
```

Comment:
What is the `StrictHostKeyChecking`?

Reply:
![image](https://github.com/user-attachments/assets/a941c5be-157c-43af-bb8e-48bb5b73789c)

Reply:
we might remove that part actually

Reply:
well we actually might need that since the instance IP address can change. What is the HOST_NAME configured as?